encoding/protobuf/protoc-gen-yarpc-go/internal/lib/lib.go

```diff
@@ -642,6 +661,43 @@ func clientServerStreamingMethods(service *protoplugin.Service) ([]*protoplugin.
 	return methods, nil
 }
 
+func fileDescriptorClosureVarName(f *protoplugin.File) (string, error) {
+	name := f.GetName()
+	if name == "" {
+		return "", fmt.Errorf("Could not create fileDescriptorClosureVarName: %s has no name", f)
+	}
+	h := sha256.Sum256([]byte(name))
+	return fmt.Sprintf("yarpcFileDescriptorClosure%s", hex.EncodeToString(h[:8])), nil
+}
+
+func encodedFileDescriptor(f *protoplugin.File) (string, error) {
+	b, err := f.SerializedFileDescriptor()
+	if err != nil {
+		return "", err
+	}
+
+	buf := make([]byte, 0, 0)
+	w := bytes.NewBuffer(buf)
+	w.WriteString("[]byte{\n")
+	for len(b) > 0 {
+		n := 16
+		if n > len(b) {
+			n = len(b)
+		}
+
+		s := ""
+		for _, c := range b[:n] {
+			s += fmt.Sprintf("0x%02x,", c)
```

Comment:
instead of string building, can we just write directly to the buffer in this loop?